### DISCUSSION TITLE
Passing extra_vars via curl does  not work

### DISCUSSION THREAD

**Discussion by MatijaTerzic** [semaphoreui/semaphore#2246] [General]
I have a task with "Survey Variables" set to "hostnames" as a string. in my playbook i have :hosts: "{{ hostnames }}"

In the semaphore logs of the task i see this:

The error was: 'hostnames' is undefined. 'hostnames' is undefined

I use this curl command:

curl -v -XPOST -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Authorization: Bearer my_token' -d '{
  "template_id": 4,
  "extra_vars": "{\"hostnames\":\"docker\"}"
}' https://semaphore_url/api/project/2/tasks

**Comment by jg1000c** [comment]
did you figure this out?

**Reply by MatijaTerzic** [reply]
Yes,i figured it out, the right syntax is:

curl -v -XPOST -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Authorization: Bearer my_token' -d '{
"template_id": 4,
"environment": "{\"hostnames\":\"docker\"}"
}' https://semaphore_url/api/project/2/tasks

**Reply by jengstrom440** [reply]
Hello, I am looking at a similar issue, and can you confirm that exact command worked in your environment (minus specific info)? Thanks!

**Reply by MatijaTerzic** [reply]
Instead of using "extra_vars" use "environment"

**Reply by jengstrom440** [reply]
Sorry, I should of clarified my question better. Can you post the _exact_ command, minus identifying info, so I can attempt to replicate? I have tried for many hours and I cannot get the "environment" to work in any circumstances, even after checking my json with online json formatters. I would be most grateful, thanks!

**Reply by MatijaTerzic** [reply]
The command is in this post...

curl ....

**Reply by jengstrom440** [reply]
The reason why I ask is it looks like that has an extra set of double quotes around the nested json "Environment" and when I put that statement inside a json checker online, they all confirm it is incorrectly formatted. You are saying that despite the formatting error, it still works?

This statement is wrong according to the online checkers:

```
{
"template_id": 4,
"environment": "{"hostnames":"docker"}"
}
```

However this is correct:

```
{
"template_id": 4,
"environment": {"hostnames":"docker"}
}
```

**Comment by jengstrom440** [comment]
I figured out my issue, for whatever reason in need backslashes in my nested JSON to work. Full explanation here https://github.com/semaphoreui/semaphore/discussions/2522  Hopefully this can help someone down the road. Thanks to @MatijaTerzic for the help.
